rsciio/utils/distributed.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2023 The HyperSpy developers
+#
+# This file is part of RosettaSciIO.
+#
+# RosettaSciIO is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# RosettaSciIO is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with RosettaSciIO. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+
+import numpy as np
+import dask.array as da
+
+
+def get_chunk_slice(
+    shape,
+    chunks="auto",
+    block_size_limit=None,
+    dtype=None,
+):
+    """
+    Get chunk slices for the :func:`rsciio.utils.distributed.slice_memmap` function.
+
+    Takes a shape and chunks and returns a dask array of the slices to be used with the
+    :func:`rsciio.utils.distributed.slice_memmap` function. This is useful for loading data from a memmaped file in a
+    distributed manner.
+
+    Parameters
+    ----------
+    shape : tuple
+        Shape of the data.
+    chunks : tuple, optional
+        Chunk shape. The default is "auto".
+    block_size_limit : int, optional
+        Maximum size of a block in bytes. The default is None.
+    dtype : numpy.dtype, optional
+        Data type. The default is None.
```

Comment:
Specify that these two are used by `da.core.normalize_chunks` when `chunks="auto"`